exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/MetricAdapter.java

```diff
@@ -224,5 +246,47 @@ private static int estimateNumSamples(int numPoints, MetricDataType type) {
     return Collections.emptyList();
   }
 
+  private static Sample createSample(
+      String name,
+      List<String> labelNames,
+      List<String> labelValues,
+      double value,
+      Optional<Exemplar> exemplar) {
+    return exemplar
+        .map(e -> new Sample(name, labelNames, labelValues, value, toPrometheusExemplar(e)))
+        .orElseGet(() -> new Sample(name, labelNames, labelValues, value));
+  }
+
+  private static io.prometheus.client.exemplars.Exemplar toPrometheusExemplar(Exemplar exemplar) {
+    // TODO: Keep filtered attributes for prometheus?
+    if (exemplar.getSpanId() != null && exemplar.getTraceId() != null) {
+      return new io.prometheus.client.exemplars.Exemplar(
+          valueOf(exemplar),
+          exemplar.getRecordTimeNanos(),
+          "traceid",
+          exemplar.getTraceId(),
+          "spanid",
+          exemplar.getSpanId());
+    }
+    return new io.prometheus.client.exemplars.Exemplar(valueOf(exemplar));
+  }
+
+  private static Optional<Exemplar> filterExemplars(
+      Collection<Exemplar> exemplars, double min, double max) {
+    return exemplars.stream()
+        .filter(exemplar -> valueOf(exemplar) <= max && valueOf(exemplar) > min)
+        .findAny();
+  }
+
+  private static double valueOf(Exemplar exemplar) {
+    double value = 0;
+    if (exemplar instanceof LongExemplar) {
+      value = ((LongExemplar) exemplar).getValue();
+    } else if (exemplar instanceof DoubleExemplar) {
+      value = ((DoubleExemplar) exemplar).getValue();
+    }
+    return value;
```

Comment:
this method makes me very sad. If only boxing into Number weren't an unacceptable cost. :(

Reply:
Agreed